src/Microsoft.DotNet.Darc/DarcLib/Helpers/ManifestHelper.cs

```diff
@@ -77,7 +82,37 @@ public static JObject GenerateDarcAssetJsonManifest(IEnumerable<DownloadedBuild>
                     nonShipping = extraAsset.Asset.NonShipping,
                     source = extraAsset.SourceLocation,
                     targets = GetTargetPaths(extraAsset),
-                    barAssetId = extraAsset.Asset.Id
+                    barAssetId = extraAsset.Asset.Id,
+                    size = extraAsset.SizeInBytes
+                }),
+            topLevelDuplicates = dropSizeReport?.DuplicatedAssets.Where(d => d.Locations.Any(l => l.optionalSubAssetPath == null))
+                                                        .OrderByDescending(d => d.TotalSize)
+                                                        .ThenBy(d => d.TotalCopies).Select(d =>
+                new
+                {
+                    totalSize = d.TotalSize,
+                    totalCopies = d.TotalCopies,
+                    originalSize = d.OriginalSize,
+                    locations = d.Locations.Select(l =>
+                        new
+                        {
+                            path = l.location,
+                            subpathInContainer = l.optionalSubAssetPath
+                        })
+                }),
+            duplicates = dropSizeReport?.DuplicatedAssets.OrderByDescending(d => d.TotalSize)
+                                                        .ThenBy(d => d.TotalCopies).Select(d =>
+                new
+                {
+                    totalSize = d.TotalSize,
+                    totalCopies = d.TotalCopies,
+                    originalSize = d.OriginalSize,
+                    locations = d.Locations.Select(l =>
+                        new
+                        {
+                            path = l.location,
+                            subpathInContainer = l.optionalSubAssetPath
+                        })
                 })
```

Comment:
There is some code duplication here, but it's minor and it's for infrastructure which has a different bar. So it can be ignored.

Reply:
I'll clean it up. I was mainly trying to write this quickly to get some reads on improvement areas.